src/main/java/io/github/thebusybiscuit/slimefun4/core/services/profiler/PerformanceSummary.java

```diff
@@ -0,0 +1,157 @@
+        sender.sendMessage(ChatColors.color("&6Active Blocks: &e" + totalTickedBlocks));
+        sender.sendMessage("");
+
+        summarizeTimings("Chunks", sender, entry -> {
+            int count = profiler.getBlocksOfId(entry.getKey());
+            String time = NumberUtils.getAsMillis(entry.getValue());
+
+            if (count > 1) {
+                String average = NumberUtils.getAsMillis(entry.getValue() / count);
+
+                return entry.getKey() + " - " + count + "x (" + time + ", " + average + " avg/block)";
+            }
+            else {
+                return entry.getKey() + " - " + count + "x (" + time + ')';
+            }
+        }, items.entrySet().stream());
+
+        sender.sendMessage("");
+
+        summarizeTimings("Blocks", sender, entry -> {
+            int count = profiler.getBlocksInChunk(entry.getKey());
+            String time = NumberUtils.getAsMillis(entry.getValue());
+
+            return entry.getKey() + " - " + count + "x (" + time + ")";
+        }, chunks.entrySet().stream());
+    }
+
+    private void summarizeTimings(String prefix, CommandSender sender, Function<Map.Entry<String, Long>, String> formatter, Stream<Map.Entry<String, Long>> stream) {
+        List<Entry<String, Long>> results = stream.sorted(Map.Entry.comparingByValue(Comparator.reverseOrder())).collect(Collectors.toList());
+
+        if (sender instanceof Player) {
+            TextComponent component = new TextComponent(prefix);
+            component.setColor(ChatColor.GOLD);
+
+            TextComponent hoverComponent = new TextComponent("\n   Hover for more details...");
+            hoverComponent.setColor(ChatColor.GRAY);
+            hoverComponent.setItalic(true);
+            StringBuilder builder = new StringBuilder();
+            int hidden = 0;
+
+            for (Map.Entry<String, Long> entry : results) {
+                if (entry.getValue() > VISIBILITY_THRESHOLD) {
+                    builder.append("\n&e").append(formatter.apply(entry));
+                }
+                else {
+                    hidden++;
+                }
+            }
+
+            builder.append("\n\n&c+ &6").append(hidden).append(" more");
+            hoverComponent.setHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, TextComponent.fromLegacyText(ChatColors.color(builder.toString()))));
+
+            component.addExtra(hoverComponent);
+            sender.spigot().sendMessage(component);
+        }
+        else {
+            int hidden = 0;
+
+            sender.sendMessage(ChatColor.GOLD + prefix);
+
+            for (Map.Entry<String, Long> entry : results) {
+                if (entry.getValue() > VISIBILITY_THRESHOLD) {
+                    sender.sendMessage("  " + ChatColor.stripColor(formatter.apply(entry)));
+                }
+                else {
+                    hidden++;
+                }
+            }
+
+            sender.sendMessage("+ " + hidden + " more");
```

Comment:
You know you can send \n in chat right? You don't need to send 30 messages. 1 message with \n.

It's great. M